docs/debugging.md

```diff
@@ -196,3 +196,100 @@ will ensure that the source files are kept as well, such files will be loaded
+```
+
+See [the porting guide](porting.md) for more detail about porting Cython code to
+work under free-threading.
+
+You may wonder why you are able to install a wheel for the current Cython
+release at all. This is because Cython ships a pure-python wheel tagged with
+`py2.py3-none-any`, which pip will install if it cannot find another wheel that
+is compatible. A future version of Cython will ship a wheel with compiled code
+that supports the free-threaded build.
+
+The current nightly wheel is a pure-python build, so it works on all
+architectures. The pure-python version of Cython is usually only marginally
+slower than a compiled version, so you should default to installing the wheel in
+CI instead of compiling Cython, which can take up to a few minutes on some CI
+runners.
+
+### Cython compilation errors arising from `static_assert`
+
+This error arises from a limitation in Cython that exposes a CPython
+implementation detail. Under clang, the issue manifests in error messages like
+the following example, generated by building scikit-learn with a C99 compilation
+environment:
+
+```
+      In file included from sklearn/__check_build/_check_build.cpython-313t-darwin.so.p/sklearn/__check_build/_check_build.pyx.c:4315:
+      In file included from /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_frame.h:13:
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:15: error: expected parameter declarator
+      static_assert(COLD_EXIT_INITIAL_VALUE > ADAPTIVE_COOLDOWN_VALUE,
+                    ^
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_backoff.h:125:33: note: expanded from macro 'COLD_EXIT_INITIAL_VALUE'
+      #define COLD_EXIT_INITIAL_VALUE 64
+                                      ^
+      In file included from sklearn/__check_build/_check_build.cpython-313t-darwin.so.p/sklearn/__check_build/_check_build.pyx.c:4315:
+      In file included from /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_frame.h:13:
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:15: error: expected ')'
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_backoff.h:125:33: note: expanded from macro 'COLD_EXIT_INITIAL_VALUE'
+      #define COLD_EXIT_INITIAL_VALUE 64
+                                      ^
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:14: note: to match this '('
+      static_assert(COLD_EXIT_INITIAL_VALUE > ADAPTIVE_COOLDOWN_VALUE,
+                   ^
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
+      static_assert(COLD_EXIT_INITIAL_VALUE > ADAPTIVE_COOLDOWN_VALUE,
+      ^
+      int
+```
+
+Freethreading support in Cython relies CPython internals. In particular, cython
```

Comment:
```suggestion
Freethreading support in Cython relies on CPython internals. In particular, cython
```

Reply:
It doesn't probably matter that much, but I guess using Free-threading (with a dash in the middle instead of in one word) may be more consistent with the rest of the repo?